cli/ts/commands/genProofs.ts

```diff
@@ -474,47 +476,75 @@ export const genProofs = async ({
-    },
-    totalSpentVoiceCredits: {
-      spent: poll.totalSpentVoiceCredits.toString(),
-      salt: asHex(tallyCircuitInputs!.newSpentVoiceCreditSubtotalSalt as BigNumberish),
-      commitment: asHex(newSpentVoiceCreditsCommitment),
-    },
-    perVOSpentVoiceCredits: {
-      tally: poll.perVOSpentVoiceCredits.map((x) => x.toString()),
-      salt: asHex(tallyCircuitInputs!.newPerVOSpentVoiceCreditsRootSalt as BigNumberish),
-      commitment: asHex(newPerVOSpentVoiceCreditsCommitment),
-    },
-  };
+  let tallyFileData: TallyData;
+
+  if (useQuadraticVoting) {
+    // Compute newPerVOSpentVoiceCreditsCommitment
+    newPerVOSpentVoiceCreditsCommitment = genTreeCommitment(
+      poll.perVOSpentVoiceCredits,
+      BigInt(asHex(tallyCircuitInputs!.newPerVOSpentVoiceCreditsRootSalt as BigNumberish)),
+      poll.treeDepths.voteOptionTreeDepth,
+    );
+
+    // Compute newTallyCommitment
+    newTallyCommitment = hash3([
+      newResultsCommitment,
+      newSpentVoiceCreditsCommitment,
+      newPerVOSpentVoiceCreditsCommitment,
+    ]);
+
+    tallyFileData = {
+      maci: maciContractAddress,
+      pollId: pollId.toString(),
+      network: network?.name,
+      chainId: network?.chainId.toString(),
+      tallyAddress: tallyContractAddress,
+      newTallyCommitment: asHex(tallyCircuitInputs!.newTallyCommitment as BigNumberish),
+      results: {
+        tally: poll.tallyResult.map((x) => x.toString()),
+        salt: asHex(tallyCircuitInputs!.newResultsRootSalt as BigNumberish),
+        commitment: asHex(newResultsCommitment),
+      },
+      totalSpentVoiceCredits: {
+        spent: poll.totalSpentVoiceCredits.toString(),
+        salt: asHex(tallyCircuitInputs!.newSpentVoiceCreditSubtotalSalt as BigNumberish),
+        commitment: asHex(newSpentVoiceCreditsCommitment),
+      },
+      perVOSpentVoiceCredits: {
+        tally: poll.perVOSpentVoiceCredits.map((x) => x.toString()),
+        salt: asHex(tallyCircuitInputs!.newPerVOSpentVoiceCreditsRootSalt as BigNumberish),
+        commitment: asHex(newPerVOSpentVoiceCreditsCommitment),
+      },
+    };
+  } else {
+    newTallyCommitment = hashLeftRight(newResultsCommitment, newSpentVoiceCreditsCommitment);
+
+    tallyFileData = {
+      maci: maciContractAddress,
+      pollId: pollId.toString(),
+      network: network?.name,
+      chainId: network?.chainId.toString(),
+      tallyAddress: tallyContractAddress,
+      newTallyCommitment: asHex(tallyCircuitInputs!.newTallyCommitment as BigNumberish),
+      results: {
+        tally: poll.tallyResult.map((x) => x.toString()),
+        salt: asHex(tallyCircuitInputs!.newResultsRootSalt as BigNumberish),
+        commitment: asHex(newResultsCommitment),
+      },
+      totalSpentVoiceCredits: {
+        spent: poll.totalSpentVoiceCredits.toString(),
+        salt: asHex(tallyCircuitInputs!.newSpentVoiceCreditSubtotalSalt as BigNumberish),
+        commitment: asHex(newSpentVoiceCreditsCommitment),
+      },
+    };
```

Comment:
Let's get rid of this duplication. You can move it outside the if and just update object props that we need to change.